diff --git a/configure b/configure
index 180f3ef0ef..0814adacc8 100755
--- a/configure
+++ b/configure
@@ -4277,7 +4277,8 @@ case $target_os in
         SLIB_INSTALL_NAME='$(SLIBNAME)'
         SLIB_INSTALL_LINKS=
         # soname not set on purpose
-        SHFLAGS=-shared
+        #SHFLAGS=-shared
+	SHFLAGS='-shared -Wl,-soname,$(SLIBNAME)'
         ;;
     haiku)
         prefix_default="/boot/common"
